plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestIgniteConnectorTest.java

```diff
@@ -0,0 +1,391 @@
+        }
+    }
+
+    @Override
+    public void testInsert()
+    {
+        String tableName = "test_insert_" + randomNameSuffix();
+        @Language("SQL") String create = "CREATE TABLE " + tableName + "(phone varchar(15) , custkey bigint , acctbal double)";
+        @Language("SQL") String query = "SELECT phone, custkey, acctbal FROM customer";
+
+        assertUpdate(create);
+        assertQuery("SELECT count(*) FROM " + tableName, "SELECT 0");
+
+        assertUpdate("INSERT INTO " + tableName + " " + query, "SELECT count(*) FROM customer");
+
+        assertQuery("SELECT * FROM " + tableName + "", query);
+
+        assertUpdate("INSERT INTO " + tableName + " (custkey) VALUES (-1)", 1);
+        assertUpdate("INSERT INTO " + tableName + " (custkey, phone) VALUES (-4, '3283-2001-01-01')", 1);
+        assertUpdate("INSERT INTO " + tableName + " (custkey, phone) VALUES (-2, '3283-2001-01-02')", 1);
+        assertUpdate("INSERT INTO " + tableName + " (phone, custkey) VALUES ('3283-2001-01-03', -3)", 1);
+        assertUpdate("INSERT INTO " + tableName + " (acctbal, custkey) VALUES (1234, -2234)", 1);
+
+        assertQuery("SELECT * FROM " + tableName + "", query
+                + " UNION ALL SELECT null, -1, null"
+                + " UNION ALL SELECT '3283-2001-01-01', -4, null"
+                + " UNION ALL SELECT '3283-2001-01-02', -2, null"
+                + " UNION ALL SELECT '3283-2001-01-03', -3, null"
+                + " UNION ALL SELECT null, -2234, 1234");
+
+        // UNION query produces columns in the opposite order
+        // of how they are declared in the table schema
+        // max custkey => 1500
+        assertUpdate(
+                "INSERT INTO " + tableName + " (custkey, phone, acctbal) " +
+                        "SELECT custkey + 1600, phone, acctbal FROM customer " +
+                        "UNION ALL " +
+                        "SELECT 100000 + custkey, phone, acctbal FROM customer",
+                "SELECT 2 * count(*) FROM customer");
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testCreateTableWithAllProperties()
+    {
+        String tableWithAllProperties = "test_create_with_all_properties";
+        assertUpdate("CREATE TABLE IF NOT EXISTS " + tableWithAllProperties + " (a bigint, b double, c varchar, d date) WITH (primary_key = ARRAY['a', 'b'])");
+    }
+
+    @Override
+    protected TestTable createTableWithDefaultColumns()
+    {
+        return new TestTable(
+                onRemoteDatabase(),
+                "public.tbl",
+                "(col_required bigint," +
+                        "col_nullable bigint," +
+                        "col_default bigint DEFAULT 43," +
+                        "col_nonnull_default bigint DEFAULT 42," +
+                        "col_required2 bigint NOT NULL, " +
+                        "dummy_id varchar NOT NULL primary key)");
+    }
+
+    @Override
+    public void testShowCreateTable()
```

Comment:
I'm not sure why this test creates a new table unlike super class. Why not use `orders` table?

Reply:
Add the `orderkey` as the `primary key` in the `tableDefinition`

Reply:
In such case, I would recommend adding a new test method instead.